joblib/parallel.py

```diff
@@ -1259,7 +1256,7 @@ def _start(self, iterator, pre_dispatch):
     def _get_outputs(self, iterator, pre_dispatch):
         """Iterator returning the tasks' output as soon as they are ready."""
         current_thread_id = threading.get_ident()
```

Comment:
Maybe rename `current_thread_id` to `dispatch_start_thread_id` to make the if condition less confusing.